test/integration/node-specific/resource_clean_up.test.ts

```diff
@@ -0,0 +1,90 @@
+import { expect } from 'chai';
+
+import { runScript } from './resource_tracking_script_builder';
+
+/**
+ * This 5MB range is selected arbitrarily and should likely be raised if failures are seen intermittently.
+ *
+ * The goal here is to catch unbounded memory growth, currently runScript defaults to 100 iterations of whatever test code is passed in
+ * More than a 5MB growth in memory usage after the script has finished and _should have_ cleaned up all its resources likely indicates that
+ * the growth will continue if the script is changed to iterate the the test code more.
+ */
+const MB_PERMITTED_OFFSET = 5;
+
+describe('Driver Resources', () => {
+  let startingMemoryUsed;
+  let endingMemoryUsed;
+  let heap;
+
+  beforeEach(function () {
+    if (globalThis.AbortController == null) {
+      if (this.currentTest) this.currentTest.skipReason = 'Need AbortController to run this test';
+      this.currentTest?.skip();
+    }
+    if (typeof this.configuration.serverApi === 'string') {
+      if (this.currentTest) {
+        this.currentTest.skipReason = 'runScript does not support serverApi settings';
+      }
+      this.currentTest?.skip();
+    }
+  });
+
+  before('create leak reproduction script', async function () {
+    if (globalThis.AbortController == null || typeof this.configuration.serverApi === 'string') {
+      return;
+    }
+    const res = await runScript(
+      'no_resource_leak_connect_close',
+      this.configuration,
+      async function run({ MongoClient, uri }) {
+        const mongoClient = new MongoClient(uri);
+        await mongoClient.connect();
+        const db = mongoClient.db();
+        await Promise.all([
```

Comment:
why `db.collections()` and why this many times? also, are we interested in ensuring no leak when there is only one client or should we test multiples? (your helper script runs multiple iterations, but I am trying to understand what the actual scenario under test is)

Reply:
The operation is arbitrary, the parallelism is to get "some" number of connections in the connection pool, this bug does reproduce without this though, I can make it one db.collections call

Reply:
ok but why db.collections? why isn't the `connect` call enough? if you just want connections in the connection pool, then setting a minpoolsize feels like a more straightforward approach

Reply:
I minimized the reproduction more, it's a connect and ping, although I guess the connect is redundant with any operation. I tried only setting a minPoolSize, but the leak stopped reproducing so it would seem that running an operation over the connections is required.

@baileympearson FYI ^